libs/barista-components/table/src/expandable/expandable-row.ts

```diff
@@ -170,6 +170,7 @@ export class DtExpandableRow extends DtRow
     elementRef: ElementRef,
   ) {
     super(elementRef);
+    this._table._hasExpandableRows = true;
```

Comment:
@ffriedl89 How is this going to work combined with the `when` condition on rows? 
If there are expandable rows should initially, and the expression driving the `when` condition changes resulting in no expandable rows shown, how does that one reset on the table?

Reply:
The only reason this property seems to exist at the moment is for styling reasons since expandable rows use a 2px padding vs the 1px padding of the default rows which messes up the alignment of the header-checkbox with those in the table rows.
Note that this problem is not yet fixed and you would have to manually adjust styling when switching between expandable/none expandable rows in that situation to ensure perfect alignment of the header checkbox with the table. 

We could avoid this altogether if left/right paddings were consistent between expandable and none-expandable rows which would be possible since the expandable rows only really need the additional pixel on the top and bottom for the 2px thick border used there in the expanded state. Not sure if that could cause any other styling issues due top/bottom padding no longer being the same as left/right padding.